clicommand/job_update.go

```diff
@@ -51,6 +51,11 @@ var JobUpdateCommand = cli.Command{
 			Usage:  "Which job should the change be made to",
 			EnvVar: "BUILDKITE_JOB_ID",
 		},
+		cli.BoolFlag{
+			Name:   "append",
```

Comment:
(changing the label of the current job is a bit of a contrived example, because you're right, you already have `$BUILDKITE_JOB_NAME` floating around)

One cool thing about this API is that it can do things to other jobs that the agent has access to. So if you know the $BUILDKITE_JOB_UUID of another job, you could append to it's label (without having to know it's current label) Another possibility is that you can "append" a new agent agent rule to a job, or even "append" a new environment variable to it.

You're right in that some attributes might not be appendable, and I can handle that in a validation error message "you can't do that" and some docs.

Reply:
Make sense! 👍🏼 I just wanted to ask, because it stood out as being a slightly awkward command line arg with the other properties, but it's not a big deal.